### PR TITLE
Added navigation component

### PR DESCRIPTION
**Pull request by ReinisZogots**

### CODE REVIEW

**Review by Jose-Hash-Dev — APPROVED** [review]

File: src/components/molecules/navigationMenu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export interface MenuItem {
+  title: string,
+  url?: string,
+  isSubMenu?: boolean,
+  subMenu?: SubMenuItem[]
+}
+
+const Menu: React.FC<MenuProps> = ({ toggleSubMenu, setIsMenuOpen}) => {
```
Comment (VSlobolinskyi) [inline src/components/molecules/navigationMenu/index.tsx:18]: Also if you are not spreading props you can go with just `const Menu = (props: MenuProps) => {}` but this is not something you should implement but rather a good to know thing

File: src/structure.json [hunk 2]
```diff
@@ -60,6 +60,22 @@
       "url": "#"
     }
   ],
+  "navigationIcons": {
```
Comment (VSlobolinskyi) [inline src/structure.json:63]: Please make a nested structure where you for example would have: 
`
"shoppingCart": {
    "altTag": "..."
    "path": "..."
}
`

**Review by VSlobolinskyi — CHANGES_REQUESTED** [review]

**Review by VSlobolinskyi — APPROVED** [review]

File: src/components/molecules/shoppingCart/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useRef } from "react";
+import Structure from "../../../structure.json";
+import "./style.scss";
+
+interface CartProps{
```
Comment (Jalal999) [inline src/components/molecules/shoppingCart/index.tsx]: space

File: src/components/molecules/shoppingCart/index.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  setIsCartOpen : React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const Cart: React.FC<CartProps> = ({setIsCartOpen}) => {
+  const cart = Structure.cart;
+  const cartRef = useRef<HTMLDivElement>(null);
+  useEffect (() => {
+    const handler = (event: Event) =>{
+      if(cartRef.current && !cartRef.current.contains(event.target as Node)){
```
Comment (Jalal999) [inline src/components/molecules/shoppingCart/index.tsx]: space

File: src/components/molecules/shoppingCart/index.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+interface CartProps{
+  setIsCartOpen : React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const Cart: React.FC<CartProps> = ({setIsCartOpen}) => {
+  const cart = Structure.cart;
+  const cartRef = useRef<HTMLDivElement>(null);
+  useEffect (() => {
+    const handler = (event: Event) =>{
```
Comment (Jalal999) [inline src/components/molecules/shoppingCart/index.tsx]: space

**Review by Jalal999 — APPROVED** [review]

**Pull request merged**